common/src/main/java/com/wynntils/features/trademarket/TradeMarketQuickSearchFeature.java

```diff
@@ -0,0 +1,139 @@
+    private static final Pattern CUT_PATTERN =
+            Pattern.compile("(Emerald Pouch|\\[.*?\\]|[\\uE000-\\uF8FF]+|\\u2B21)\\s*");
+
+    private static final Pattern POTION_PATTERN = Pattern.compile("^Potion of (. \\w+)");
+    private static final Pattern EOL_PATTERN = Pattern.compile("À+$");
+
+    private static final int SEARCH_SLOT = 47;
+    private String searchQuery;
+    private boolean inTradeMarket = false;
+    private boolean inSearchChat = false;
+    private boolean quickSearching = false;
+
+    @SubscribeEvent
+    public void onScreenOpen(ScreenOpenedEvent.Post event) {
+        if (!(Models.Container.getCurrentContainer() instanceof TradeMarketContainer)) {
+            inTradeMarket = false;
+            quickSearching = false;
+            return;
+        }
+        inTradeMarket = true;
+    }
+
+    @SubscribeEvent
+    public void onScreenClosed(ScreenClosedEvent event) {
+        if (!inSearchChat || !(event.getScreen() instanceof ChatScreen)) return;
+        if (autoCancel.get() && KeyboardUtils.isKeyDown(GLFW.GLFW_KEY_ESCAPE)) {
+            McUtils.sendChat("cancel");
+        }
+        inSearchChat = false;
+    }
+
+    // EventPriority.HIGH so that InventoryEmeraldCountFeature does not render.
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void MenuClosedEvent(MenuClosedEvent event) {
+        if (!inTradeMarket) return;
+        inTradeMarket = false;
+        if (!quickSearching) return;
+        event.setCanceled(true);
+        quickSearching = false;
+    }
+
+    private void tryQuickSearch(Slot hoveredSlot) {
+        if (!inTradeMarket || hoveredSlot == null || !hoveredSlot.hasItem()) return;
+        quickSearching = true;
+        searchQuery =
+                StyledText.fromComponent((hoveredSlot.getItem().getHoverName())).getStringWithoutFormatting();
+        searchQuery = getSearchQuery(searchQuery);
+        if (searchQuery == null || searchQuery.isBlank()) return;
+        ContainerUtils.clickOnSlot(
+                SEARCH_SLOT,
+                McUtils.containerMenu().containerId,
+                GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                McUtils.containerMenu().getItems());
+    }
+
+    // EventPriority.HIGH so that this takes priority over TradeMarketAutoOpenChatFeature.hidePrompt.
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void onChatMessageReceive(ChatMessageReceivedEvent event) {
+        if (!quickSearching || !event.getOriginalStyledText().stripAlignment().matches(TYPE_TO_CHAT_PATTERN)) return;
+        if (!instantSearch.get() || KeyboardUtils.isKeyDown(GLFW.GLFW_KEY_LEFT_SHIFT)) {
+            McUtils.mc().setScreen(new ChatScreen(searchQuery));
+            if (hidePrompt.get()) {
+                event.setCanceled(true);
+            }
+            inSearchChat = true;
+        } else {
+            event.setCanceled(true);
+            McUtils.sendChat(searchQuery);
+        }
+    }
```

Comment:
This doesn't need to high priority as the default will be above the low of the hide prompt option but it's fine. But do please put this method above `tryQuickSearch` as we usually put the @SubscribeEvent methods first

Reply:
Fixed [c52aaa4](https://github.com/Wynntils/Wynntils/pull/2908/commits/c52aaa467e843e76de65393e510514b171dda91b)